src/index/generateTrees.ts

```diff
@@ -6,8 +6,9 @@ import { RootOption as Root } from "./shared/RootOption";
 import { buildGraph } from "./generateTrees/buildGraph";
 import { findEntryPoints } from "./generateTrees/findEntryPoints";
 import { toFractalTree } from "./generateTrees/toFractalTree";
+import { detectLonelyFiles } from "./shared/detect-lonely-files";
 
-export function generateTrees(restructureMap: { [key: string]: string[] }) {
+export function generateTrees(restructureMap: { [key: string]: string[] }, renameOnlyChildFlag: boolean) {
```

Comment:
I think we would benefit for putting all these options and flags in an object instead of another argument in the function. It would be easier to add more options/flags later.

```suggestion
export function generateTrees(restructureMap: { [key: string]: string[] }, opts: { renameOnlyChildFlag: boolean }) {
```